### PR TITLE
Update debug docs

### PR DESCRIPTION
**Pull request by Veetaha**
Updated `debugging.md` file, added two useful demo videos.

### CODE REVIEW

**Comment by Veetaha** [comment]
cc @SomeoneToIgnore

File: docs/dev/debugging.md [hunk 1]
```diff
@@ -1,44 +1,67 @@
...
 
-```
-$ git clone https://github.com/rust-analyzer/rust-analyzer.git --depth 1
-$ cd rust-analyzer
-$ code .
-```
+- Install all TypeScript dependencies
+  ```bash
+  cd editors/code
+  npm i
```
Comment (lnicola) [inline docs/dev/debugging.md]: Maybe `npm install` for us plebs :-).

File: docs/dev/debugging.md [hunk 2]
```diff
@@ -1,44 +1,67 @@
...
+  ```
 
-- To attach to the `lsp server` in linux you'll have to run:
+## Common knowledge
+
+* All debug configurations open new `[Extension Development Host]` VSCode instance
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
* All debug configurations open a new `[Extension Development Host]` VSCode instance
 where **only** the `rust-analyzer` extension being debugged is enabled. 
```

File: docs/dev/debugging.md [hunk 3]
```diff
@@ -1,44 +1,67 @@
...
+
+
+## Debug TypeScript VSCode extension
 
-  `echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope`
+- `Run Extension` - runs the extension with globally installed `ra_lsp_server` binary.
+- `Run Extension (Dev Server)` - runs extension with binary from your locally built `target/debug/ra_lsp_server`.
 
-  This enables ptrace on non forked processes
+TypeScript debugging is configured to watch your source edits and recompile.
```
Comment (lnicola) [inline docs/dev/debugging.md]: TIL

File: docs/dev/debugging.md [hunk 4]
```diff
@@ -1,44 +1,67 @@
...
+
+* All debug configurations open new `[Extension Development Host]` VSCode instance
+where **only** your `rust-analyzer` extension is enabled.
+* To activate the extension you need to open any Rust project folder in `[Extension Development Host]`.
+
+
+## Debug TypeScript VSCode extension
 
-  `echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope`
+- `Run Extension` - runs the extension with globally installed `ra_lsp_server` binary.
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
- `Run Extension` - runs the extension with the globally installed `ra_lsp_server` binary.
```

File: docs/dev/debugging.md [hunk 5]
```diff
@@ -1,44 +1,67 @@
...
+- `Run Extension (Dev Server)` - runs extension with binary from your locally built `target/debug/ra_lsp_server`.
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
- `Run Extension (Dev Server)` - runs extension with the locally built LSP server (`target/debug/ra_lsp_server`).
```

File: docs/dev/debugging.md [hunk 6]
```diff
@@ -1,44 +1,67 @@
...
 
-  This enables ptrace on non forked processes
+TypeScript debugging is configured to watch your source edits and recompile.
+To apply changes to already running debug process press <kbd>Ctrl+Shift+P</kbd> and run the following command in your `[Extension Development Host]`
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
To apply changes to an already running debug process press <kbd>Ctrl+Shift+P</kbd> and run the following command in your `[Extension Development Host]`
```

File: docs/dev/debugging.md [hunk 7]
```diff
@@ -1,44 +1,67 @@
...
 
-- Ensure the dependencies for the extension are installed, run the `npm: install - editors/code` task in vscode.
+```
+> Developer: Reload Window
+```
+
+## Debug Rust LSP server
+
+- To attach to the `lsp server` in linux you'll have to run:
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
- When attaching a debugger to an already running `rust-analyzer` server on Linux you might need to run:
```

File: docs/dev/debugging.md [hunk 8]
```diff
@@ -1,44 +1,67 @@
...
 
-- Launch the `Debug Extension`, this will build the extension and the `lsp server`.
+  ```
+  echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope
+  ```
 
-- A new instance of vscode with `[Extension Development Host]` in the title.
+  This enables ptrace on non-forked processes
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
```

File: docs/dev/debugging.md [hunk 9]
```diff
@@ -1,44 +1,67 @@
...
 
-  Don't worry about disabling `rls` all other extensions will be disabled but this one.
+- Enable debug symbols in `Cargo.toml`:
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
- By default, the LSP server is built without debug information. To enable it, you'll need to change `Cargo.toml`:
```

File: docs/dev/debugging.md [hunk 10]
```diff
@@ -1,44 +1,67 @@
...
+  ```toml
+    [profile.dev]
+    debug = 2
+  ```
 
-- In the new vscode instance open a rust project, and navigate to a rust file
+- Select `Run Extension (Dev Server)` to run your locally built `target/debug/ra_lsp_server`.
 
-- In the original vscode start an additional debug session (the three periods in the launch) and select `Debug Lsp Server`.
+- In the original VSCode window once again select `Attach To Server` debug configuration.
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
- In the original VSCode window once again select the `Attach To Server` debug configuration.
```

File: docs/dev/debugging.md (hunk 1 above)
Comment (Veetaha) [reply docs/dev/debugging.md]: Might you propose a change here?

Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
  npm install
```

File: docs/dev/debugging.md (hunk 2 above)
Comment (Veetaha) [reply docs/dev/debugging.md]: Thank you, articles are my weak point :D

File: docs/dev/debugging.md [hunk 11]
```diff
@@ -1,44 +1,67 @@
...
 
-- A list of running processes should appear select the `ra_lsp_server` from this repo.
+- A list of running processes should appear. Select the `ra_lsp_server` from this repo.
 
 - Navigate to `crates/ra_lsp_server/src/main_loop.rs` and add a breakpoint to the `on_task` function.
 
 - Go back to the `[Extension Development Host]` instance and hover over a rust variable and your breakpoint should hit.
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
- Go back to the `[Extension Development Host]` instance and hover over a Rust variable and your breakpoint should hit.
```

File: docs/dev/debugging.md (hunk 1 above)
Comment (Veetaha) [reply docs/dev/debugging.md]: I mean "commit suggestion", though, ah, I'll do it)

File: docs/dev/debugging.md (hunk 2 above)
Comment (lnicola) [reply docs/dev/debugging.md]: I had no idea that Russian and most related languages had no articles :smile:.

File: docs/dev/debugging.md (hunk 1 above)
Comment (lnicola) [reply docs/dev/debugging.md]: I made one, but it doesn't show up on this page.

File: docs/dev/debugging.md (hunk 2 above)
Comment (Veetaha) [reply docs/dev/debugging.md]: TYL )

File: docs/dev/debugging.md (hunk 1 above)
Comment (Veetaha) [reply docs/dev/debugging.md]: Saw that one, already accepted!

File: docs/dev/debugging.md [hunk 12]
```diff
@@ -1,44 +1,67 @@
 # Debugging vs Code plugin and the Language Server
 
-**NOTE:** the information here is mostly obsolete
+## Prerequsities
```
Comment (lnicola) [inline docs/dev/debugging.md]:
```suggestion
## Prerequisites
```

File: docs/dev/debugging.md (hunk 2 above)
Comment (Veetaha) [reply docs/dev/debugging.md]: I suppose you can select multiple lines to edit, because we need to amend the second line it commit suggsetion too

Comment (lnicola) [reply docs/dev/debugging.md]: Sorry, I'll let you do that locally. I tried it a couple of PRs ago and it didn't turn out too well.

File: docs/dev/debugging.md [hunk 13]
```diff
@@ -1,44 +1,66 @@
-# Debugging vs Code plugin and the Language Server
+# Debugging VSCode plugin and the language server
 
-**NOTE:** the information here is mostly obsolete
+## Prerequisites
 
-Install [LLDB](https://lldb.llvm.org/) and the [LLDB Extension](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb).
+- Install [LLDB](https://lldb.llvm.org/) and the [LLDB Extension](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb).
```
Comment (kjeremy) [inline docs/dev/debugging.md]: Does anyone have luck using this extension on windows? It doesn't really work well for me.

Comment (Veetaha) [reply docs/dev/debugging.md]: What do you use instead?

Comment (edwin0cheng) [reply docs/dev/debugging.md]: In Windows I use https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools

Comment (lnicola) [reply docs/dev/debugging.md]: It's author says it works: https://www.reddit.com/r/rust/comments/f1qsx9/debugging_rust_in_vscode_in_2020/fh8f1gx/.

**Pull request merged**